core/capabilities/webapi/trigger_test.go

```diff
@@ -0,0 +1,260 @@
+
+	// var triggerRegistrationConfig *values.Map
+	triggerRegistrationConfig, err := values.NewMap(map[string]interface{}{
+		"RateLimiter":    rateLimitConfig,
+		"AllowedSenders": addresses,
+		"AllowedTopics":  topics,
+		"RequiredParams": []string{"bid", "ask"},
+	})
+	return triggerRegistrationConfig, err
+}
+func setup(t *testing.T) testHarness {
+	privateKey, _ := testutils.NewPrivateKeyAndAddress(t)
+	registry := registrymock.NewCapabilitiesRegistry(t)
+	connector := gcmocks.NewGatewayConnector(t)
+	lggr := corelogger.TestLogger(t)
+	config := ""
+
+	trigger, err := NewTrigger(config, registry, connector, privateKey, lggr)
+	require.NoError(t, err)
+
+	return testHarness{
+		registry:  registry,
+		connector: connector,
+		lggr:      lggr,
+		config:    config,
+		trigger:   trigger,
+	}
+}
+
+func gatewayRequest(t *testing.T, topics string) *api.Message {
+	privateKey := "65456ffb8af4a2b93959256a8e04f6f2fe0943579fb3c9c3350593aabb89023f"
+	messageID := "12345"
+	methodName := "web_trigger"
+	donID := "workflow_don_1"
+
+	key, err := crypto.HexToECDSA(privateKey)
+	require.NoError(t, err)
+
+	payload := `{
+         "trigger_id": "web-trigger@1.0.0",
+          "trigger_event_id": "action_1234567890",
+          "timestamp": 1234567890,
+          "topics": ` + topics + `,
+					"params": {
+						"bid": "101",
+						"ask": "102"
+					}
+        }
+`
+	payloadJSON := []byte(payload)
+	msg := &api.Message{
+		Body: api.MessageBody{
+			MessageId: messageID,
+			Method:    methodName,
+			DonId:     donID,
+			Payload:   json.RawMessage(payloadJSON),
+		},
+	}
+	err = msg.Sign(key)
+	require.NoError(t, err)
+
+	return msg
+}
+
+func TestTriggerExecute(t *testing.T) {
+	th := setup(t)
+	ctx := testutils.Context(t)
+	Config, _ := workflowTriggerConfig(th, []string{address1}, []string{"daily_price_update", "ad_hoc_price_update"})
+	triggerReq := capabilities.TriggerRegistrationRequest{
+		TriggerID: triggerID1,
+		Metadata: capabilities.RequestMetadata{
+			WorkflowID:    workflowID1,
+			WorkflowOwner: owner1,
+		},
+		Config: Config,
+	}
+	channel, err := th.trigger.RegisterTrigger(ctx, triggerReq)
+	require.NoError(t, err)
+
+	Config2, err := workflowTriggerConfig(th, []string{address1}, []string{"daily_price_update2", "ad_hoc_price_update"})
+	require.NoError(t, err)
+
+	triggerReq2 := capabilities.TriggerRegistrationRequest{
+		TriggerID: triggerID2,
+		Metadata: capabilities.RequestMetadata{
+			WorkflowID:    workflowID1,
+			WorkflowOwner: owner1,
+		},
+		Config: Config2,
+	}
+	channel2, err := th.trigger.RegisterTrigger(ctx, triggerReq2)
+	require.NoError(t, err)
+
+	t.Run("happy case single topic to single workflow", func(t *testing.T) {
+		gatewayRequest := gatewayRequest(t, `["daily_price_update"]`)
+
+		th.connector.On("SendToGateway", mock.Anything, mock.Anything, mock.Anything).Return(nil).Once()
+		th.trigger.HandleGatewayMessage(ctx, "gateway1", gatewayRequest)
+
+		sent := <-channel
```

Comment:
Checking that there is an event on first channel, but should also check that there _isn't_ one on the second.